File: persistence/src/main/java/io/sparkled/persistence/stage/impl/query/GetStagePropByUuidQuery.java
package io.sparkled.persistence.stage.impl.query;

import io.sparkled.model.entity.StageProp;
import io.sparkled.persistence.PersistenceQuery;
import io.sparkled.persistence.QueryFactory;

import java.util.Optional;
import java.util.UUID;

public class GetStagePropByUuidQuery implements PersistenceQuery<Optional<StageProp>> {

    private final int stageId;
    private final UUID uuid;

    public GetStagePropByUuidQuery(int stageId, UUID uuid) {
        this.stageId = stageId;
        this.uuid = uuid;
    }

    @Override
    public Optional<StageProp> perform(QueryFactory queryFactory) {
        StageProp sequenceChannel = queryFactory
                .selectFrom(qStageProp)
                .where(qStageProp.stageId.eq(stageId).and(qStageProp.uuid.eq(uuid)))
                .fetchFirst();

        return Optional.ofNullable(sequenceChannel);
    }
}

File: persistence/src/main/kotlin/io/sparkled/persistence/stage/impl/query/GetStagePropByUuidQuery.kt
package io.sparkled.persistence.stage.impl.query

import io.sparkled.model.entity.QStageProp.Companion.stageProp
import io.sparkled.model.entity.StageProp
import io.sparkled.persistence.PersistenceQuery
import io.sparkled.persistence.QueryFactory
import java.util.UUID

class GetStagePropByUuidQuery(private val stageId: Int, private val uuid: UUID) :
    PersistenceQuery<StageProp?> {

    override fun perform(queryFactory: QueryFactory): StageProp? {
        return queryFactory
            .selectFrom(stageProp)
            .where(stageProp.stageId.eq(stageId).and(stageProp.uuid.eq(uuid)))
            .fetchFirst()
    }
}
